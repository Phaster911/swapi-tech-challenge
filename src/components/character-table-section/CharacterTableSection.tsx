import styled from "styled-components";
import { Person } from "../../react-query/queries/use-get-people.hook";
import CharacterTable from "./character-table/CharacterTable";

const StyledSection = styled.section`
  flex-grow: 1;
`;

export type CharacterTableSectionProps = {
  people: Person[];
};

export default function CharacterTableSection({
  people,
}: CharacterTableSectionProps) {
  return (
    <StyledSection aria-labelledby="table-caption">
      <CharacterTable people={people} />
    </StyledSection>
  );
}
