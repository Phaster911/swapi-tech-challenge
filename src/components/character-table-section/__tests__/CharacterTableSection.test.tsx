import { render } from "@testing-library/react";
import { Person } from "../../../react-query/queries/use-get-people.hook";
import CharacterTableSection, {
  CharacterTableSectionProps,
} from "../CharacterTableSection";

vi.mock("../character-table/CharacterTable", () => ({
  default: () => <div>CharacterTable</div>,
}));

const person: Person = {
  id: "1",
  name: "name-1",
  gender: "gender-1",
  birthYear: "year-1",
  mass: "mass-1",
  filmsEntered: 0,
  height: "height-1",
};

describe("<Header />", () => {
  let props: CharacterTableSectionProps;
  beforeEach(() => {
    props = {
      people: [person],
    };
  });
  it("renders the component", () => {
    const { container } = render(<CharacterTableSection {...props} />);
    expect(container).toMatchSnapshot();
  });
});
