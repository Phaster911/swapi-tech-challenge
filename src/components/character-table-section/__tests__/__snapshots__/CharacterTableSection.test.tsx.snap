// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > renders the component 1`] = `
.c0 {
  flex-grow: 1;
}

<div>
  <section
    aria-labelledby="table-caption"
    class="c0"
  >
    <div>
      CharacterTable
    </div>
  </section>
</div>
`;
